src/aleph/vm/controllers/firecracker/instance.py

```diff
@@ -168,7 +164,11 @@ def _get_hostname(self) -> str:
     def _encode_user_data(self) -> bytes:
         """Creates user data configuration file for cloud-init tool"""
 
-        ssh_authorized_keys = self.resources.message_content.authorized_keys or []
+        ssh_authorized_keys: list[str] | None
+        if settings.USE_DEVELOPER_SSH_KEYS:
```

Comment:
We should ensure that we only allow developer SSH keys if they come from the CLI argument, not from the environment variables.

```suggestion
        if settings.USE_DEVELOPER_SSH_KEYS == ALLOW_DEVELOPER_SSH_KEYS:
```

Reply:
We are already doing it here:
https://github.com/aleph-im/aleph-vm/blob/main/src/aleph/vm/orchestrator/cli.py#L308

Reply:
Just for that, there we set the value, and only here we use it, so if at the moment to set it we do the job, here we don't need to check it again, because if not the `USE_DEVELOPER_SSH_KEYS` variable can be removed and only use `ALLOW_DEVELOPER_SSH_KEYS` variable to make the checks.

Reply:
Move to https://github.com/aleph-im/aleph-vm/issues/506